server/app/models/Version.java

```diff
@@ -122,7 +123,7 @@ public Optional<Program> getProgramByName(String name) {
    * can exist in a version.
    */
   public Optional<Question> getQuestionByName(String name) {
-    return getQuestions().stream()
+    return VersionRepository.getQuestionsForVersion(this).stream()
```

Comment:
**issue (non-blocking):** I missed it in the last PR, but FYI this method will need to become an instance method rather than static in order to consume the cache.

In general it's best to avoid public static methods since they require changes to all consumers when they end up needing state from the class they're associated with.

Reply:
Ah you're right, sorry for missing that too! I think that means I'll have to do a refactor that involves changing these static methods to instance methods as well, since those call the getQuestionsForVersion function

Reply:
Ended up making that change as part of this PR, so it was a larger refactor.